File: src/Services/UserManagement/UserManagement.API/Services/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using UserManagement.API.Models;

namespace UserManagement.API.Services
{
    public class UserService:IUserService
    {
        private readonly IUserRepository _repository;

        public UserService(IUserRepository userRepository)
        {
            _repository = userRepository;
        }

        public User RegisterUser(User user)
        {
            var result = _repository.RegisterUser(user);
            return (result != null) ? result : null;
        }

        public bool DeleteUser(int userId)
        {
            return _repository.DeleteUser(userId);
        }

        public IEnumerable<User> GetAllUsers()
        {
            return _repository.GetAllUsers();
        }

        public User GetUserById(int vehicleId)
        {
            return _repository.GetUserById(vehicleId);
        }

        public bool UpdateUser(int userId, User user)
        {
            return _repository.UpdateUser(userId, user);
        }
    }
}

File: src/WebSPA/src/app/services/user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { User } from '../models/user';
import { AuthenticationService } from './authentication.service';
import { Userinfo } from '../models/userinfo';

@Injectable({
  providedIn: 'root',
})
export class UserService {
  result: boolean;
  private bearerToken: string;
  constructor(private http: HttpClient, public authService: AuthenticationService) {
    this.bearerToken = this.authService.getBearerToken();
  }
  baseUrl: string = 'http://localhost:8085/api/user/';

  getUserById(userid: string) {
    return this.http.get<User>(this.baseUrl + userid, { headers: new HttpHeaders().set('Authorization', `Bearer ${this.bearerToken}`) });
  }
  deleteUser(id: string) {
    return this.http.delete<User[]>(this.baseUrl + id, { headers: new HttpHeaders().set('Authorization', `Bearer ${this.bearerToken}`) });
  }
  registerUser(user: User) {
    var userinfo: Userinfo = {
      userId: user.userId,
      username: user.username,
      password: user.password,
      fullname: user.fullname,
      contact: '',
      role: 'user',
    }
    this.result = true;

    this.authService.registerUser(userinfo)
      .subscribe(data => {
      },
        error => {
          if (error.status === 403 || error.status === 409 || error.status === 500) {
            this.result = false;
          }
        });
    if (this.result) {
      return this.http.post(this.baseUrl, user);
    }
  }
  updateUser(user: User) {
    return this.http.put(this.baseUrl + user.userId, user, { headers: new HttpHeaders().set('Authorization', `Bearer ${this.bearerToken}`) });
  }

}
